modules/reindex/src/test/java/org/elasticsearch/index/reindex/RethrottleTests.java

```diff
@@ -174,6 +171,39 @@ private void testCase(AbstractBulkByScrollRequestBuilder<?, ?> request, String a
                 response.getBatches(), greaterThanOrEqualTo(numSlices));
     }
 
+    private ListTasksResponse rethrottleTask(TaskId taskToRethrottle, float newRequestsPerSecond) {
+        // the task isn't ready to be rethrottled until it has figured out how many slices it will use. if we rethrottle when the task is
+        // in this state, the request will fail. so we try a few times
+        long start = System.nanoTime();
+        List<Throwable> failures = new ArrayList<>();
+        do {
```

Comment:
It is pretty similar; I used it at first but it only catches AssertionError, and the throwable that causes this problem is an ElasticsearchException. Think it's worth changing assertBusy or adding a similar method to ESTestCase?